evap/results/templates/results_evaluation_detail_questionnaires.html

```diff
@@ -14,17 +16,25 @@ <h4>{{ questionnaire_result.questionnaire.public_name }}</h4>
             </div>
             <h5{% if not forloop.first %} class="mt-3"{% endif %}>{{ question_result.question.text }}</h5>
             <div class="container-fluid container-striped{% if not forloop.last %} mb-3{% endif %}">
-        {% elif question_result.question.is_rating_question %}
+        {% elif question_result|is_of_type:'RatingResult' %}
             <div class="row print-no-break">
                 <div class="col-question col-lg-broad col-lg-7 my-auto">{{ question_result.question.text }}</div>
                 <div class="col-answer col-lg-broad col-lg-5 my-auto">
                     {% include 'question_result_widget.html' with question_result=question_result num_voters=evaluation.num_voters %}
                 </div>
-            </div>
-        {% elif question_result.question.is_text_question %}
-            <div class="text-answer row">
-                <div class="col-question col-lg-broad col-lg-4">{{ question_result.question.text }}</div>
-                <div class="col-answer col-lg-broad col-lg-8 d-flex">
+            {% if not question_result.has_additional_text_result %}
+                </div>
+            {% endif %}
+        {% elif question_result|is_of_type:'TextResult' %}
+            {% if not question_result.is_additional_text_result %}
+                <div class="row">
+            {% endif %}
```

Comment:
I think the first step could be to replace `is_of_type` filters with static methods if possible, imo `is_text_result` and `is_rating_result` would be sensible here.

I couldn’t try it locally, but do I interpret it correctly that the question text is doubled if a rating question has additional text answers? If yes, why?

A more higher view proposol would be to _somehow_ implement a more tree-oriented structure. I mean, currently we have headers which divide the linear-looking question results into multiple sections, which feels more like two layers instead of one. Similary, we now put results for the same question in the same row. So maybe a reshuffling of the template data into: sections (with header) → question → results. Or the merging of additional text answers into the `RatingResult` helps, idk

Reply:
i now used a different approach that should make this template much easier to read